### PR TITLE
qa: wait for daemons to come up via cephadm

### PR DESCRIPTION
**Pull request by batrick**
Rather than waiting for a set amount of time.

Fixes: https://tracker.ceph.com/issues/49684
Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by liewegas — APPROVED** [review]

File: qa/tasks/ceph_manager.py [hunk 1]
```diff
@@ -1545,10 +1545,13 @@ def flush_pg_stats(self, osds, no_wait=None, wait_for_mon=300):
     def flush_all_pg_stats(self):
         self.flush_pg_stats(range(len(self.get_osd_dump())))
 
-    def do_rados(self, remote, cmd, check_status=True):
+    def do_rados(self, cmd, pool=None, ns=None, remote=None, **kwargs):
```
Comment (tchaikov) [inline qa/tasks/ceph_manager.py:1548]:
```
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b6eecfe89a4363750d696160843a096384b357a8/teuthology/run_tasks.py", line 91, in run_tasks
    manager = run_one_task(taskname, ctx=ctx, config=config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b6eecfe89a4363750d696160843a096384b357a8/teuthology/run_tasks.py", line 70, in run_one_task
    return task(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_c7a0c64e16ba061c83f7490040af3a838e5e01ca/qa/tasks/repair_test.py", line 293, in task
    repair_test_1(manager, mdataerr, choose_primary, "scrub")
  File "/home/teuthworker/src/git.ceph.com_ceph-c_c7a0c64e16ba061c83f7490040af3a838e5e01ca/qa/tasks/repair_test.py", line 87, in repair_test_1
    manager.do_put(pool, 'repair_test_obj', '/etc/hosts')
  File "/home/teuthworker/src/git.ceph.com_ceph-c_c7a0c64e16ba061c83f7490040af3a838e5e01ca/qa/tasks/ceph_manager.py", line 1608, in do_put
    check_status=False
  File "/home/teuthworker/src/git.ceph.com_ceph-c_c7a0c64e16ba061c83f7490040af3a838e5e01ca/qa/tasks/ceph_manager.py", line 1568, in do_rados
    pre.extend(cmd)
TypeError: 'Remote' object is not iterable
```

https://pulpito.ceph.com/kchai-2021-03-11_07:16:51-rados-wip-kefu-testing-2021-03-11-1238-distro-basic-smithi/5955811/

**Review by tchaikov — CHANGES_REQUESTED** [review]

**Comment by batrick** [comment]
@tchaikov yes sorry. I'm in the middle of fixing this PR after finding out `qa: wait for daemons to come up via cephadm` was not the only change necessary. I'll switch this to a draft.

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20210312.020610

**Comment by batrick** [comment]
@tchaikov this is ready for review/qa

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2021-03-15_22:16:56-fs-wip-pdonnell-testing-20210315.182203-distro-basic-smithi/

- https://tracker.ceph.com/issues/49842
- https://tracker.ceph.com/issues/49466
- https://tracker.ceph.com/issues/49822
- https://tracker.ceph.com/issues/49240
- https://tracker.ceph.com/issues/48773
- https://tracker.ceph.com/issues/45434
- https://tracker.ceph.com/issues/49500
- https://tracker.ceph.com/issues/49843
- https://tracker.ceph.com/issues/49845
- https://tracker.ceph.com/issues/48805
- https://tracker.ceph.com/issues/49605

and failure caused by PR: https://github.com/ceph/ceph/pull/39969

**Review by tchaikov — APPROVED** [review]

File: qa/tasks/ceph_manager.py [hunk 2]
```diff
@@ -1703,67 +1710,49 @@ def rados_write_objects(self, pool, num_objects, size,
...
-            fname
-        ]
+        args = ['put', obj, fname]
         return self.do_rados(
-            self.controller,
             args,
-            check_status=False
+            check_status=False,
+            pool=pool,
+            namespace=namespace
```
Comment (tchaikov) [inline qa/tasks/ceph_manager.py:1731]:
```
2021-03-21T08:58:15.236 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_c48eb744081d22bc82d7d099d4edb67ae02551e0/teuthology/run_tasks.py", line 91, in run_tasks
    manager = run_one_task(taskname, ctx=ctx, config=config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_c48eb744081d22bc82d7d099d4edb67ae02551e0/teuthology/run_tasks.py", line 70, in run_one_task
    return task(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_1bd75a5080e9c1eb0a279260ba2c7847ef58e03d/qa/tasks/repair_test.py", line 291, in task
    repair_test_1(manager, mdataerr, choose_primary, "scrub")
  File "/home/teuthworker/src/git.ceph.com_ceph-c_1bd75a5080e9c1eb0a279260ba2c7847ef58e03d/qa/tasks/repair_test.py", line 85, in repair_test_1
    manager.do_put(pool, 'repair_test_obj', '/etc/hosts')
  File "/home/teuthworker/src/git.ceph.com_ceph-c_1bd75a5080e9c1eb0a279260ba2c7847ef58e03d/qa/tasks/ceph_manager.py", line 1731, in do_put
    namespace=namespace
  File "/home/teuthworker/src/git.ceph.com_ceph-c_1bd75a5080e9c1eb0a279260ba2c7847ef58e03d/qa/tasks/ceph_manager.py", line 1702, in do_rados
    **kwargs
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_c48eb744081d22bc82d7d099d4edb67ae02551e0/teuthology/orchestra/remote.py", line 215, in run
    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
TypeError: run() got an unexpected keyword argument 'namespace'
```

https://pulpito.ceph.com/kchai-2021-03-20_13:57:49-rados-wip-kefu2-testing-2021-03-20-1320-distro-basic-smithi/

**Review by tchaikov — CHANGES_REQUESTED** [review]

File: qa/tasks/ceph_manager.py (hunk 2 above)
Comment (batrick) [reply qa/tasks/ceph_manager.py:1731]: Fixed.

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2021-03-24_23:26:35-fs-wip-pdonnell-testing-20210324.190252-distro-basic-smithi/

- https://tracker.ceph.com/issues/49500
    qa: "Assertion `cb_done' failed."
- https://tracker.ceph.com/issues/50019
    qa: mount failure with cephadm "probably no MDS server is up?"
- https://tracker.ceph.com/issues/50020
    qa: "RADOS object not found (Failed to operate read op for oid cephfs_mirror)"
- https://tracker.ceph.com/issues/48773
    qa: scrub does not complete
- https://tracker.ceph.com/issues/45434
    qa: test_full_fsync (tasks.cephfs.test_full.TestClusterFull) failed
- https://tracker.ceph.com/issues/48805
    mds: "cluster [WRN] Scrub error on inode 0x1000000039d (/client.0/tmp/blogbench-1.0/src/blogtest_in) see mds.a log and `damage ls` output for details"
- https://tracker.ceph.com/issues/48772
    qa: pjd: not ok 9, 44, 80
- https://tracker.ceph.com/issues/50021
    qa: snaptest-git-ceph failure during mon thrashing
- https://tracker.ceph.com/issues/48771
    qa: iogen: workload fails to cause balancing
- https://tracker.ceph.com/issues/50016
    qa: test_damage: "RuntimeError: 2 mutations had unexpected outcomes"
- https://tracker.ceph.com/issues/49466
    qa: "Command failed on gibba030 with status 1: 'set -ex\nsudo dd of=/tmp/tmp.ZEeZBasJer'"

**Comment by batrick** [comment]
@tchaikov CephFS is good. Please merge when you're ready.

**Comment by tchaikov** [comment]
@batrick ack. will include this changeset in my next batch.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-27_12:17:59-rados-wip-kefu-testing-2021-03-27-1545-distro-basic-gibba/
- https://pulpito.ceph.com/kchai-2021-03-28_03:01:25-rados-wip-kefu-testing-2021-03-27-1545-distro-basic-smithi/

failures

- https://tracker.ceph.com/issues/49344
- https://tracker.ceph.com/issues/49260
- https://tracker.ceph.com/issues/49633

**Pull request merged**